service-discovery/src/main/scala/com/lightbend/rp/servicediscovery/scaladsl/ServiceLocator.scala

```diff
@@ -18,44 +18,58 @@ package com.lightbend.rp.servicediscovery.scaladsl
 
 import akka.actor._
 import akka.pattern.ask
-import com.lightbend.dns.locator.{ ServiceLocator => DnsServiceLocator, Settings => DnsServiceLocatorSettings }
+import com.lightbend.dns.locator.{ServiceLocator => DnsServiceLocator, Settings => DnsServiceLocatorSettings}
 import com.lightbend.rp.common._
 import java.net.URI
+import java.util.concurrent.ThreadLocalRandom
 import scala.concurrent.Future
 
 case class ServiceLocator(as: ActorSystem) {
   def lookup(name: String): Future[Option[URI]] = ServiceLocator.lookup(name)(as)
 }
 
 object ServiceLocator {
-  def lookup(name: String)(implicit as: ActorSystem): Future[Option[URI]] =
-    Platform.active match {
+  def lookup(name: String)(implicit as: ActorSystem): Future[Option[URI]] = {
+    val settings = Settings(as)
+
+    settings.externalServiceAddresses.get(name) match {
+      case Some(services) =>
+        Future.successful(Some(services(ThreadLocalRandom.current.nextInt(services.length))))
```

Comment:
I'd suggest allowing the caller to perform the service address selection themselves, i.e.

```
case class ServiceLocator {
  import ServiceLocator._
  
  def lookup(name: String, addressSelection: AddressSelection = AddressSelectionRandom): Future[Option[URI]] = 
    ServiceLocator.lookup(name, addressSelection)(as)
  // ...
}

object ServiceLocator {
  type AddressSelection = Seq[URI] => Option[URI]
  val AddressSelectionRandom: AddressSelection = (services => services(ThreadLocalRandom.current.nextInt(services.length))) 

  def lookup(name: String, addressSelection: AddressSelection = AddressSelectionRandom)(implicit as: ActorSystem): Future[Option[URI]] = {
    // ...
```

Then this line can be:

```
Future.successful(Some(addressSelection(services))))
```

Reply:
Good idea - I did this for the Java API as well.